src/Http/Controllers/CategoryController.php

```diff
@@ -2,169 +2,42 @@
 
 namespace App\Http\Controllers;
 
+use App\Core\Exception\RecordNotFoundException;
+
+require base_path("./Core/Exceptions/RecordNotFoundException.php");
+
 class CategoryController
 {
-
     public function show(int $id)
     {
-        foreach ($this->getCategories() as $category) {
-            if ($category['id'] === $id) {
-                view("showCategory", [
-                    "products" => $category['Products'],
-                    "categoryName" => $category['CategoryName'],
-                    "categories" => $this->getCategories()
-                ]);
-                die();
-            }
+        if ($this->getCategory($id) != false) {
```

Comment:
No need for the if statement since the `getCategory` method throws an exception if the category is not found (query result equals `false`).